functions/src/project/delete-project.ts

```diff
@@ -40,3 +41,27 @@ export const onDeleteProjectRepositories: CloudFunction<DocumentSnapshot> = fire
     }
 
   });
+
+export const onDeleteProject: CloudFunction<DocumentSnapshot> = firestore
+  .document('projects/{projectUid}')
+  .onDelete(async (projectSnapshot: DocumentSnapshot, context: EventContext) => {
+
+    try {
+      const project: DocumentData = projectSnapshot.data();
+
+      if (Array.isArray(project.monitors) && project.monitors.length > 0) {
+        const promises: Promise<WriteResult>[] = [];
+
+        for (const monitor of project.monitors) {
```

Comment:
I think this can be all removed if instead of using `deleteMonitorPings`, `deleteProjectPings` is used because it has the loop already?

Reply:
Good catch .